File: 0242-valid-anagram/0242-valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        if (s.size() != t.size())return false;
        unordered_map<char , int>smp , tmp;
        for (int i = 0 ; i < s.size(); i++)smp[s[i]]++;
        for (int i = 0 ; i < t.size(); i++)tmp[t[i]]++;
        for (int i = 0 ; i < s.size(); i++)
        {
            if (tmp[s[i]] != smp[s[i]])return false;
        }
        return true;
    }
};

File: 0242-valid-anagram/0242-valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        if(s.length() != t.length())return false;
        Map<Character, Integer> mp = new HashMap<>();
        for (char c : s.toCharArray())
        {
            if (mp.containsKey(c))
            {
                mp.put(c , mp.get(c)+1);
            }
            else 
                mp.put(c , 1);
        }
        for (char c : t.toCharArray())
        {
             if (mp.containsKey(c) && mp.get(c) > 0)
            {
                mp.put(c , mp.get(c)-1);
            }
            else return false;
        }
        return true;
    }
}